doc/development/06_ca_manager.md

```diff
@@ -0,0 +1,978 @@
+    /// and all relevant content will be withdrawn from the repository.
+    pub async fn ca_parent_remove(
+        &self,
+        handle: Handle,
+        parent: ParentHandle,
+        actor: &Actor
+    ) -> KrillResult<()> { ... }
+
+    /// Updates a parent of a CA, this can be used to update the service uri and/or
+    /// identity certificate for an existing parent.
+    pub async fn ca_parent_update(
+        &self,
+        handle: Handle,
+        parent: ParentHandle,
+        contact: ParentCaContact,
+        actor: &Actor,
+    ) -> KrillResult<()> { ... }
+
+    /// Returns the parent statuses for this CA
+    pub async fn ca_parent_statuses(&self, ca: &Handle) -> KrillResult<ParentStatuses> { ... }
+
+    /// Refresh all CAs:
+    /// - process all CAs in parallel
+    /// - process all parents for CAs in parallel
+    ///    - send pending requests if present, or
+    ///    - ask parent for updates and process if present
+    pub async fn cas_refresh_all(&self, actor: &Actor) { ... }
+
+    /// Synchronizes a CA with one of its parents:
+    ///   - send pending requests if present; otherwise
+    ///   - get and process updated entitlements
+    ///
+    /// Note: if new request events are generated as a result of processing updated entitlements
+    ///       then they will trigger that this synchronization is called again so that the pending
+    ///       requests can be sent.
+    pub async fn ca_sync_parent(&self, handle: &Handle, parent: &ParentHandle, actor: &Actor) -> KrillResult<()> { ... }
+}
+```
+
+
+CA as Parent Related Functions
+------------------------------
+
+The following functions are used to manage children of CAs.
+
+```rust
+/// # CAs as children
+///
+impl CaManager {
+    /// Adds a child under a CA. The 'service_uri' is used here so that
+    /// the appropriate `ParentCaContact` can be returned. If the `AddChildRequest`
+    /// contains resources not held by this CA, then an `Error::CaChildExtraResources`
+    /// is returned.
+    pub async fn ca_add_child(
+        &self,
+        ca: &Handle,
+        req: AddChildRequest,
+        service_uri: &uri::Https,
+        actor: &Actor,
+    ) -> KrillResult<ParentCaContact> { ... }
+
+    /// Show details for a child under the TA.
+    pub async fn ca_show_child(
+        &self,
+        ca: &Handle,
+        child: &ChildHandle
+    ) -> KrillResult<ChildCaInfo> { ... }
+
+    /// Gets an RFC8183 Parent Response for the child.
+    pub async fn ca_parent_response(
+        &self,
+        ca: &Handle,
+        child_handle: ChildHandle,
+        tag: Option<String>,
+        service_uri: &uri::Https,
+    ) -> KrillResult<rfc8183::ParentResponse> { ... }
+
+    /// Update a child under this CA. The submitted `UpdateChildRequest` can contain a
+    /// new `IdCert`, or `ResourceSet`. If both are updated in a single update, then
+    /// an `Error::CaChildUpdateOneThing` is returned. When resource entitlements are updated,
```

Comment:
Is CaChildUpdateOneThing returned because it is technically not possible to update both at once, or because the RFCs forbid it, or because Krill just hasn't implemented support for it yet, or because we could but don't want to support it, or ...?

Reply:
It was like this because the CLI will only submit a single thing, and there is no good reason to update both at the same time that I can think of.

Then again, I am not against it either. Or we could have separate endpoints and comments for this if we want to make the split explicit.

I just documented what we have, but I am happy to change the behaviour. Do you have a preference?

Reply:
No preference, just that the documentation naturally led to the question "why not?" and thus I saw an opportunity to capture the reason here rather than leave the reader to wonder.

Reply:
I inclined to say: let's remove the restriction and have one end-point for updating various aspects of a Child CA. So, we could also add other things in future, if we want.

The alternative is to change the API and potentially end up with a lot of finegrained functions. Not bad in itself, but maybe noisy?

I am really unsure what it is best to be honest. But removing the error is probably good.

Reply:
Oh and splitting would be a breaking change in the API, so if no one else has a strong preference then that is a good reason to keep it as-is (but remove the error, we **can** just update two things).

Reply:
Pushed.. allowing both. I forgot to include a comment, but -- theoretically updating the ID cert could fail and it's tried first. However, this is extremely unlikely - it can only really fail if the child does not exist. On the other hand if the updates failed on resources the ID would have already been updated. I don't think this is an issue really, but something to keep in mind when using this.